fe1-web/Model/Network/JsonRpcRequest.ts

```diff
@@ -8,74 +8,74 @@ import { ProtocolError } from './ProtocolError';
  */
 export class JsonRpcRequest {
 
-    public readonly method: JsonRpcMethod;
-    public readonly id?: number;
-    public readonly params: JsonRpcParams;
+  public readonly method: JsonRpcMethod;
+  public readonly id?: number;
+  public readonly params: JsonRpcParams;
 
-    static fromJson(jsonString: string) : JsonRpcRequest {
-        // validate with ajv (json-schema)
+  static fromJson(jsonString: string) : JsonRpcRequest {
+    // validate with ajv (json-schema)
 
-        let obj = JSON.parse(jsonString);
+    let obj = JSON.parse(jsonString);
 
-        return new JsonRpcRequest(obj as JsonRpcRequest);
-    }
-
-    constructor(req: Partial<JsonRpcRequest>) {
-        Object.assign(this, req);
+    return new JsonRpcRequest(obj as JsonRpcRequest);
+  }
 
-        if (!req.method) {
-            throw new ProtocolError("Undefined 'method' in JSON-RPC");
-        }
-        if (req.params === undefined || req.params === null) {
-            throw new ProtocolError("Undefined 'params' in JSON-RPC");
-        }
+  constructor(req: Partial<JsonRpcRequest>) {
+    Object.assign(this, req);
 
-        this.method = req.method;
-        this.id = req.id || undefined;
-        this.params = this._parseParams(req.params);
+    if (!req.method) {
+      throw new ProtocolError("Undefined 'method' in JSON-RPC");
+    }
+    if (req.params === undefined || req.params === null) {
+      throw new ProtocolError("Undefined 'params' in JSON-RPC");
     }
 
-    public verify(): boolean {
-        switch(this.method) {
-            // notification methods, expect no ID
-            case JsonRpcMethod.BROADCAST:
-                if ( this.id !== undefined ) {
-                    return false;
-                }
-                break;
+    this.method = req.method;
+    this.id = req.id || undefined;
+    this.params = this._parseParams(req.params);
+  }
 
-            // RPC methods, expect an ID
-            case JsonRpcMethod.PUBLISH:
-            case JsonRpcMethod.SUBSCRIBE:
-            case JsonRpcMethod.UNSUBSCRIBE:
-            case JsonRpcMethod.CATCHUP:
-                if ( this.id === undefined ) {
-                    return false;
-                }
-                break;
+  public verify(): boolean {
```

Comment:
I seem to remember you had convinced me of dropping the verify() and just verifying these properties in the constructor, right?

Reply:
I indeeed forgot the top layer of the protocol ^^'. Working on it